Modules/signalmodule.c

```diff
@@ -1893,7 +1911,15 @@ PyErr_SetInterruptEx(int signum)
 
     signal_state_t *state = &signal_global_state;
     PyObject *func = get_handler(signum);
-    if (func != state->ignore_handler && func != state->default_handler) {
+    int is_ign = PyObject_RichCompareBool(func, state->ignore_handler, Py_EQ);
```

Comment:
Is it okay to call a Python code in `PyErr_SetInterruptEx()`?

If `func` is a custom callable, it can have an `__eq__` method which can call an arbitrary Python code here.

It may be safer to use `PyLong_Check()` + `PyLong_GetLongAndOverflow()`.

Reply:
Good point!

I took a different approach and added a helper function. We can safely assume that the internal handler objects are always exact longs. Comparison between exact longs should never fail.